lib/check-each-pcb-trace-non-overlapping/check-each-pcb-trace-non-overlapping.ts

```diff
@@ -107,6 +106,15 @@ export function checkEachPcbTraceNonOverlapping(
     if (segmentA.x1 === segmentA.x2 && segmentA.y1 === segmentA.y2) continue
 
     for (const obj of nearbyObjects) {
+      // ignore obstacles on wrong layer (except vias)
+      if (
+        obj.type !== "pcb_via" &&
+        "layer" in obj &&
+        obj.layer !== segmentA.layer
+      ) {
+        continue
+      }
```

Comment:
can we make this more explicit, e.g.
```suggestion
    const obstacleLayers = getLayersOfPcbElement(obj)
    if (obstacleLayers.includes(segmentA.layer)) continue
```

```tsx
export function getLayersOfPcbElement(obj: AnyPcbElement) {
  if (obj.type === "pcb_smtpad") {
   // ...
  }
} 
```

I don't like the `"layer" in obj` check because it's kind of not explicit and doesn't give us good future handling for buried or blind vias